src/marqo/s2_inference/hf_utils.py

```diff
@@ -16,15 +24,95 @@ def __init__(self, *args, **kwargs) -> None:
 
-        self.model = AutoModelForSentenceEmbedding(self.model_name).to(self.device)
-        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+        model_location_presence = ModelProperties.model_location in self.model_properties
+        path = self.model_properties.get("localpath", None) or self.model_properties.get("url", None)
+        # HF models can be loaded from 3 entries: path (url or localpath), model_name, or model_location
+        if (path is not None) + (self.model_name is not None) + (model_location_presence is True) != 1:
+            raise InvalidModelPropertiesError("Exactly one of `url`, `localpath` or `model_location`, `name` can be specified"
+                                              " in `model_properties` for `hf` models as they conflict with each other in model loading."
+                                              " Please ensure that exactly one of these is specified in `model_properties` and retry.")
+        elif path is not None:
+            if validators.url(path) is True:
+                self.model_path = download_model(url = path, download_dir=ModelCache.hf_cache_path)
+            elif os.path.isdir(path) or os.path.isfile(path):
+                self.model_path = path
+        elif self.model_name is not None:
+            # Loading from structured huggingface repo directly, token is required directly
+            self.model_path = self.model_name
+            self.model = AutoModelForSentenceEmbedding(self.model_path).to(self.device)
+            self.tokenizer = AutoTokenizer.from_pretrained(self.model_path)
+            return
+        elif model_location_presence is not None:
+            # This is a special case for huggingface models, where we can load a model directory from a repo
+            if ("hf" in self.model_properties["model_location"]) and ("repo_id" in self.model_properties["model_location"]["hf"]) and \
+                ("filename" not in self.model_properties["model_location"]["hf"]):
+                return self._load_from_private_hf_repo()
+            else:
+                self.model_path = self._download_from_repo()
 
+        else:
+            raise InvalidModelPropertiesError(
+                "Exactly one of `url`, `localpath` or `model_location`, `name` must be specified"
+                " in `model_properties` for `hf` models as they conflict with each other in model loading."
+                " Please ensure that exactly one of these is specified in `model_properties` and retry.")
+
+        # We need to do extraction here if necessary
+        self.model_path = validate_huggingface_archive(self.model_path)
+        self.model = AutoModelForSentenceEmbedding(self.model_path).to(self.device)
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_path)
+
+    def _load_from_private_hf_repo(self) -> None:
+        """
+        Load a private model from a huggingface repo directly using the `name` attribute in `model_properties`
```

Comment:
Is this correct? I see no reference of `name` in this function. Also it uses a bunch of other variables.

Reply:
Updated code, we use `repo_id` instead of `name` now. 
resolved